spyder/plugins/projects/plugin.py

```diff
@@ -188,18 +189,14 @@ def on_editor_available(self):
         treewidget.sig_renamed.connect(self.editor.renamed)
         treewidget.sig_tree_renamed.connect(self.editor.renamed_tree)
         treewidget.sig_module_created.connect(self.editor.new)
-        treewidget.sig_file_created.connect(
-            lambda t: self.editor.new(text=t))
+        treewidget.sig_file_created.connect(self._new_editor)
+
+        self.sig_project_loaded.connect(self._editor_open_files)
+        self.sig_project_closed[bool].connect(self._editor_open_files)

```

Comment:
Sorry, after realizing that the `_editor_open_files` method is connected to `sig_project_loaded` and `sig_project_closed`, I think it makes more sense to call it `_setup_files_in_editor`.